Cython/Compiler/ExprNodes.py

```diff
@@ -2115,6 +2116,8 @@ def analyse_types(self, env):
             from . import Buffer
             Buffer.used_buffer_aux_vars(entry)
         self.analyse_rvalue_entry(env)
+        if entry.is_cpp_optional:
+            self.dereference_cpp_optional = True
```

Comment:
1) Shouldn't `analyse_rvalue_entry()` do this?
2) Do we really (really!) need a new and completely option specific field in `NameNode` (and `AttributeNode`, as it seems) for this? Or is it information that we can infer from some existing state? (It's an rvalue, it's cpp-optional, so …)

Reply:
> Or is it information that we can infer from some existing state? (It's an rvalue, it's cpp-optional, so …)

In principle the `.is_target` attribute should tell us that it's a rvalue. In practice that doesn't look to be set (ever). I've changed it so that `.is_target` is actually set and intered it as you suggest.